### PR TITLE
fix: display correct info regarding trains/night buses on 24 hour pass

### PR DESCRIPTION
**Pull request by mortennordseth**
Fixes https://github.com/AtB-AS/kundevendt/issues/2247 

Updated information regarding trains and night buses for 24 hour pass. It was previously the same as for period tickets.  

Single tickets were not available in staging during the time of testing and I was therefore not able to verify that the information is still correct for other ticket types. 

<details>
<summary>Screenshots</summary>

Details screen for 24h pass before: 
sim_before

Details screen for 24h pass after:
sim_after

</details>

### CODE REVIEW

**Comment by mortennordseth** [comment]
Verified today that the information is still correct on single tickets.

File: src/screens/Ticketing/Ticket/Details/DetailsScreen.tsx [hunk 1]
```diff
@@ -83,8 +91,10 @@ export default function DetailsScreen({navigation, route}: Props) {
         {shouldShowValidTrainTicketNotice && (
           <MessageBox
             message={
-              isSingleTicket(firstTravelRight)
+              preassignedFareProduct?.type === 'single'
                 ? t(PurchaseOverviewTexts.samarbeidsbillettenInfo.single)
+                : preassignedFareProduct?.type === 'hour24'
+                ? t(PurchaseOverviewTexts.samarbeidsbillettenInfo.hour24)
                 : t(PurchaseOverviewTexts.samarbeidsbillettenInfo.period)
```
Comment (rosvik) [inline src/screens/Ticketing/Ticket/Details/DetailsScreen.tsx:98]: I find these double "`x ? a : b`-statements" to be kind of hard to read. Could we extract this to a function instead? Something like

```ts
const getTrainTicketNoticeText = (fareProductType: string) => {
  if (fareProductType === 'single') return t(PurchaseOverviewTexts.samarbeidsbillettenInfo.single);
  if (fareProductType === 'hour24') return t(PurchaseOverviewTexts.samarbeidsbillettenInfo.hour24);
  return t(PurchaseOverviewTexts.samarbeidsbillettenInfo.period);
}
```

**Review by rosvik — APPROVED** [review]
Looks great 🙌 Just a code-style-nitpick below.

File: src/screens/Ticketing/Ticket/Details/DetailsScreen.tsx (hunk 1 above)
Comment (mortennordseth) [reply src/screens/Ticketing/Ticket/Details/DetailsScreen.tsx:98]: I agree! I'll do that, thanks 👏

**Review by rosvik — APPROVED** [review]
💪

**Pull request merged**